src/simtools/testing/sim_telarray_metadata.py

```diff
@@ -0,0 +1,164 @@
+_logger = logging.getLogger(__name__)
+
+
+def assert_sim_telarray_metadata(file, array_model):
+    """
+    Assert consistency of sim_telarray metadata with given array model.
+
+    Parameters
+    ----------
+    file: Path
+        Path to the sim_telarray file.
+    array_model: ArrayModel
+        Array model to compare with.
+    """
+    global_meta, telescope_meta = read_sim_telarray_metadata(file)
+    _logger.info(f"Found metadata in sim_telarray file for {len(telescope_meta)} telescopes")
+    site_parameter_mismatch = _assert_model_parameters(global_meta, array_model.site_model)
+
+    if len(telescope_meta) != len(array_model.telescope_model):
+        raise ValueError(
+            f"Number of telescopes in sim_telarray file ({len(telescope_meta)}) does not match "
+            f"number of telescopes in array model ({len(array_model.telescope_model)})"
+        )
+
+    for telescope_name in array_model.telescope_model.keys():
+        if not get_sim_telarray_telescope_id(telescope_name, file):
+            raise ValueError(f"Telescope {telescope_name} not found in sim_telarray file metadata")
+
+    telescope_parameter_mismatch = [
+        _assert_model_parameters(telescope_meta[i], model)
+        for i, (_, model) in enumerate(array_model.telescope_model.items(), start=1)
+    ]
+
+    # ensure printout of all mismatches, not only those found first
+    if len(site_parameter_mismatch) > 0:
+        raise ValueError(
+            f"Site model parameters do not match sim_telarray metadata: "
+            f"{site_parameter_mismatch} "
+            f"telescope_parameter_mismatch: {telescope_parameter_mismatch}"
+        )
+    if any(len(m) > 0 for m in telescope_parameter_mismatch):
+        raise ValueError(
+            f"Telescope model parameters do not match sim_telarray metadata: "
+            f"{telescope_parameter_mismatch}"
+        )
+
+
+def _assert_model_parameters(metadata, model):
+    """
+    Assert that model parameter values matches the values in the sim_telarray metadata.
+
+    Parameters
+    ----------
+    metadata: dict
+        Metadata dictionary.
+    model: SiteModel, TelescopeModel
+        Model to compare with.
+
+    Returns
+    -------
+    invalid_parameter_list: list
+        List of parameters that do not match.
+
+    """
+    config_reader = SimtelConfigReader()
+
+    invalid_parameter_list = []
+
+    for param in model.parameters:
+        sim_telarray_name = _sim_telarray_name_from_parameter_name(param)
+        if sim_telarray_name in metadata.keys():
+            parameter_type = model.parameters[param]["type"]
+            if parameter_type not in ("string", "dict", "boolean"):
+                value, _ = config_reader.extract_value_from_sim_telarray_column(
+                    [metadata[sim_telarray_name]], parameter_type
+                )
+            else:
+                value = metadata[sim_telarray_name]
+                value = (int)(value) if value.isnumeric() else value
```

Comment:
Why is it always cast to `int` here? Don't we have some floats in the configuration?

Reply:
OK, that is indeed a bit confusing (and made me think of negative integers as well), but this code works, so no complaints.